### PR TITLE
fix(core): fix SIGSEGV on writing to table with intensive row cancel and rollback operations

### PR DESCRIPTION
**Pull request by ideoma**
When `TableWriter` cancels rows or rollbacks rows it can move file mapped page backward.
If copy uncommitted is called afterwards it can receive out of date mapped address in `MemoryM.map` call.

The fix is to update `absolutePointer` every time `jump()` changes the mapped page.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryPARWImpl.java|5|5|100.00%|

File: core/src/test/java/io/questdb/cairo/vm/MemoryPMARImplTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Test
+    public void testJumpChangesActivePage() throws Exception {
+        TestUtils.assertMemoryLeak(() -> {
+            long pageSize = Files.PAGE_SIZE;
+            ConcurrentLinkedQueue<Throwable> allErrors = new ConcurrentLinkedQueue<>();
+            ObjList<Thread> threads = new ObjList<>();
+            FilesFacade ff = FilesFacadeImpl.INSTANCE;
+
+            for (int thread = 0; thread < 10; thread++) {
```
[the remark this answers is not in the document]
Comment (ideoma) [reply core/src/test/java/io/questdb/cairo/vm/MemoryPMARImplTest.java:55]: It's counterintuitive but the it's not reproducible with single thread.

If you map / unmap in single threaded manner page 0 OS always return you same pointer value.

File: core/src/test/java/io/questdb/cairo/vm/MemoryPMARImplTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class MemoryPMARImplTest {
+    private static final Log LOG = LogFactory.getLog(MemoryPMARImplTest.class);
+
+    @ClassRule
+    public static TemporaryFolder temp = new TemporaryFolder();
+
+    @Test
+    public void testJumpChangesActivePage() throws Exception {
```
[the remark this answers is not in the document]
Comment (ideoma) [reply core/src/test/java/io/questdb/cairo/vm/MemoryPMARImplTest.java:48]: It's specific to `MemoryM` and `MemoryPARWImpl` doesn't support M(apped).

[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/cairo/vm/MemoryPMARImplTest.java:48]: I understand that the problem is mmap-related. I was asking whether the jumps and assertions that we do in this test are valuable on their own - probably they're not.

**Pull request merged**